PR: FlexLayout demo for Flex Basis

bbrennanbasis opened the pull request:
This PR closes #73.
Created a demo to demonstrate the `setItemBasis()` method for `FlexLayout`.

hyyan commented on src/main/code_snippets/flexlayout/Basis.txt:
```diff
@@ -0,0 +1,11 @@
+Button basisButton = new Button("Set basis", e -> {
```
@bbrennanbasis we are not using snippets anymore. It should be removed

MatthewHawkins replied:
@hyyan Just to verify, are we making snippets totally obsolete? If so, we should remove them, and the ability to expand/condense code from all demos.

hyyan replied:
Yes, Matthew, they are obsolete. I believe I recently opened an issue for this, and they should be removed. Maintaining them is too time-consuming and prone to errors.

hyyan commented:
@bbrennanbasis Smart, I like it. One thing I’d suggest is that it’s not entirely clear what the user is supposed to interact with in the demo. I missed the button selection part. What’s the plan for this? Will you add instructions directly in the demo, or are you planning to update the documentation to explain how the demo works?

hyyan commented on src/main/java/com/webforj/samples/views/flexlayout/FlexBasisView.java:
```diff
@@ -0,0 +1,84 @@
+
+@Route
+@FrameTitle("Flex Basis")
+public class FlexBasisView extends Composite<Div> {
```
Refactor this method to reduce its cognitive complexity. This means that not everything should go into the constructor. Move the event listeners to a separate method to help lower the complexity.

hyyan replied:
https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=94&id=webforj_webforj-docs-samples&open=AZK6TfL8zADAUu5a4nOK

bbrennanbasis commented:
@hyyan I've added your revision requests. Additionally, I've added tool tip text to the main button that changes based on how many boxes are selected.

MatthewHawkins reviewed (changes requested):
@bbrennanbasis Hey Ben - looks great! I just ran it myself and also missed that I was supposed to click on a box first. I would add something, either to the docs, or to the demo (the docs are probably the best place) mentioning what to do in the demo.

I don't have any overt changes for this program - go ahead and make a small addition to the FlexLayout article on the doc site saying something like "The above (or the following) demo allows you to select a box and change its basis...). Once you've done that, ask us for review again! 👍

hyyan commented:
@bbrennanbasis  An explanation of how the demo works still needs to be added to the Flex Layout documentation.

bbrennanbasis commented:
@hyyan Added the explanation and the URL in webforj-docs [PR #295](https://github.com/webforj/webforj-docs/pull/295).

hyyan commented:
@MatthewHawkins, please review this and check if it aligns with the plans for documenting the new configuration approach. It might be best to hold off until you've completed that.
